viennats/LSlib/output.hpp

```diff
@@ -452,6 +472,588 @@ namespace lvlset {
 
         f.close();
     }
-}
+
+
+    //used to test the systems endianess, which is needed for levelset file io
+    union {
+      uint16_t shortVar;    // binary  number of length 16 Bits
+      uint8_t  charVar[2];  // 2 binary numbers, each 8 Bits
+    } test_endianness;
+
+    bool bigEndian(){
+      test_endianness.shortVar = 0x8000; // MSB of 16
+      return test_endianness.charVar[0] != 0;
+    }
+
+    template <class GridTraitsType> GridTraitsType get_grid_traits_from_lvst_file(std::string path){
+      //reads the grid information from a levelset file
+
+      std::ifstream fin(path);
+      if(!fin.is_open()) msg::print_error("ERROR: Couldn't open the file: " + path);
+      char buff[9] = {};
+      fin.read(buff, 9);
+      //Comparing Identification Bytes
+      if(std::string(buff).compare(0, 4, "LvSt")) msg::print_error("File is not a levelset file.");
+      if(LVST_FILE_VERSION_NUMBER !=  buff[4]-48) msg::print_error("File version does not match!");
+      if(bigEndian() != buff[5]-48) msg::print_warning("File was written in a different byte order than it is being read. Results may be incorrect!");
+      const int dim = buff[6]-48;
+      const int D = GridTraitsType::dimensions;
+
+      double grid_delta;
+      long long grid_min[D] = {};
+      long long grid_max[D] = {};
+      char b_conditions[D] = {};
+
+      int bytes_per_grid_limit = buff[8]-48;
+      //since grid limits can be negative we need a mask to fill the remaining bytes with FF if the sign bit is set (Two's complement)
+      unsigned long long sign_bit_mask = 0;
+      --sign_bit_mask <<= bytes_per_grid_limit * CHAR_BIT;
+
+      //read in the grid properties,which have variable sizes
+      for(int i=dim;i--;){
+        fin.read((char *)&grid_min[i], bytes_per_grid_limit);
+        //if the sign bit is set, fill the remaining bits with 1s(Two's complement)
+        if(grid_min[i] >> (bytes_per_grid_limit * CHAR_BIT-1)) grid_min[i] |= sign_bit_mask;
+        fin.read((char *)&grid_max[i], bytes_per_grid_limit);
+        if(grid_max[i] >> (bytes_per_grid_limit * CHAR_BIT-1)) grid_max[i] |= sign_bit_mask;
+        fin.read((char *)&b_conditions[i], 1);
+      }
+      fin.read((char *)&grid_delta, sizeof(double));
+
+#ifdef VERBOSE
+      msg::print_message("\nReading grid properties from levelset file..." + path);
+      std::ostringstream oss;
+      oss << "Bytes per grid limit: " << (int)bytes_per_grid_limit << std::endl;
+      for(int i=dim;i--;){
+        oss << "Dimension " << i << ":" << std::endl
+            << "    Grid min: " << grid_min[i] << std::endl
+            << "    Grid max: " << grid_max[i] << std::endl
+            << "    Boundary condition: " << (int)b_conditions[i] << std::endl;
+      }
+      oss << "Grid delta: " << grid_delta;
+      msg::print_message(oss.str());
+#endif
+      if(fin.fail()) msg::print_error("Couldn't read grid properties from file: " + path);
```

Comment:
if(fin.fail()) msg::print_error("Could not read grid properties from file: " + path);